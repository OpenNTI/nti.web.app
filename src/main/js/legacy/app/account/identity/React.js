const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
