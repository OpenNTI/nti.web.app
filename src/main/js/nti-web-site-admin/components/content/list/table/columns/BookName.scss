@import '~@nti/style-common';

.content-table-container {
	.bookname-col {
		padding-left: 1rem;

		.cell {
			display: flex;

			.book-icon {
				width: 5rem;
			}

			.book-info {
				max-width: calc(100% - 150px);
				font-size: 16px;
				line-height: 19px;
				color: $primary-grey;
				white-space: nowrap;
				margin-left: 0.5rem;

				.title {
					overflow: hidden;
					text-overflow: ellipsis;
				}
			}
		}
	}
}
