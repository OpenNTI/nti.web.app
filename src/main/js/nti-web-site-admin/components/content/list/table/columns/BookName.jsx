import React from 'react';
import PropTypes from 'prop-types';
import {scoped} from '@nti/lib-locale';
import {LinkTo} from '@nti/web-routing';
import {Presentation} from '@nti/web-commons';

const t = scoped('nti-web-site-admin.components.content.list.table.columns.BookName', {
	headerTitle: 'Type',
	title: 'Name'
});

export default class BookName extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired
	}

	static cssClassName = 'bookname-col';

	static Name = () => t('title')

	static SortKey = 'title'

	render () {
		const {item} = this.props;
		return (
			<LinkTo.Path to={`./book/${item.getID()}`} activeClassName="active" exact>
				<div className="cell">
					<Presentation.Asset contentPackage={item} type="landing">
						<img className="book-icon"/>
					</Presentation.Asset>
					<div className="book-info">
						<div className="title">{item.title}</div>
					</div>
				</div>
			</LinkTo.Path>
		);
	}
}
