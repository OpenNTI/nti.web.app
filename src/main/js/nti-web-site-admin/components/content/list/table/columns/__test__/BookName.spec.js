/* eslint-env jest */
import React from 'react';
import { create, act } from 'react-test-renderer';

import BookName from '../BookName';

describe('Site admin content table book name column test', () => {
	test('Test name', () => {
		let cmp;
		act(() => {
			cmp = create(
				<BookName
					item={{
						title: 'book name',
						getID: () => 'bookName',
						getDefaultAssetRoot() {
							return 'testRoot';
						},
					}}
				/>
			);
		});

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
		cmp.unmount();
	});
});
