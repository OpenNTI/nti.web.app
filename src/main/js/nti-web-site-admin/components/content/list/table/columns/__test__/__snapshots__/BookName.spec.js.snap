// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin content table book name column test Test name 1`] = `
<div
  className="nti-link-to-path"
  exact={true}
  to="./book/bookName"
>
  <div
    className="cell"
  >
    <img
      className="book-icon"
      onError={[Function]}
      src="contentpackage-landing-232x170.png?t=0"
    />
    <div
      className="book-info"
    >
      <div
        className="title"
      >
        book name
      </div>
    </div>
  </div>
</div>
`;
